app/src/main/kotlin/me/myshows/android/ui/activities/CommentsActivity.kt

```diff
@@ -0,0 +1,193 @@
+package me.myshows.android.ui.activities
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.os.Build
+import android.os.Bundle
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.text.Html
+import android.text.format.DateUtils
+import android.util.Pair
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import kotlinx.android.synthetic.main.comment_layout.view.*
+import kotlinx.android.synthetic.main.comments_activity.*
+import me.myshows.android.MyShowsApplication
+import me.myshows.android.R
+import me.myshows.android.api.MyShowsClient
+import me.myshows.android.model.Comment
+import me.myshows.android.model.EpisodeComments
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import java.util.*
+
+private const val ROOT_COMMENTS = 0
+private const val INITIAL_NESTING_LEVEL = 1
+
+private const val COMMENT_BOUND = 25
+
+const val EPISODE_ID = "episodeId"
+const val EPISODE_TITLE = "episodeTitle"
+
+class CommentsActivity : HomeActivity() {
+
+    private val TAG = CommentsActivity::class.java.simpleName
+
+    private lateinit var client: MyShowsClient
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.comments_activity)
+
+        client = MyShowsApplication.getMyShowsClient(this)
+
+        setupActionBar(toolbar)
+        setupRecyclerView(recycler_view)
+
+        val episodeId = extractAndBindEpisodeData(intent)
+        val offset = resources.getDimensionPixelSize(R.dimen.default_padding)
+        loadData(episodeId, offset)
+    }
+
+    @SuppressLint("PrivateResource")
+    private fun setupRecyclerView(recyclerView: RecyclerView) {
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.setHasFixedSize(true)
+    }
+
+    private fun extractAndBindEpisodeData(intent: Intent): Int {
+        supportActionBar?.title = intent.getStringExtra(EPISODE_TITLE)
+        return intent.getIntExtra(EPISODE_ID, 0)
+    }
+
+    private fun loadData(episodeId: Int, offset: Int) {
+        client.comments(episodeId)
+                .compose(bindToLifecycle())
+                .observeOn(Schedulers.computation())
+                .map { extractOrderedComments(it) }
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { recycler_view.adapter = CommentsAdapter(it, offset) }
+    }
+
+    private fun extractOrderedComments(information: EpisodeComments): List<Pair<Comment, Int>> {
```

Comment:
You can create data class instead of `Pair<Comment, Int>` to specify `Int` field meaning